zip-0308.rst

```diff
@@ -0,0 +1,326 @@
+* it is easier to finish in a reasonable length of time.
+
+The choice of 500 blocks as the batch interval ensures that each batch occurs
+at a different time of day, which may help to mitigate problems with the
+availability of nodes being correlated with the local time-of-day.
+
+Simulations show that the migration process will typically complete reasonably
+quickly even if the amount to be migrated is large:
+
++-----------+--------------------------------------------+
+|           |     Time in days to complete migration     |
+|   Amount  +-----------------+--------+-----------------+
+|           | 10th-percentile | median | 90th-percentile |
++===========+=================+========+=================+
+|      1 ZEC|       1.01      |  1.50  |       2.33      |
++-----------+-----------------+--------+-----------------+
+|     10 ZEC|       1.63      |  2.46  |       3.45      |
++-----------+-----------------+--------+-----------------+
+|    100 ZEC|       2.46      |  3.91  |       5.72      |
++-----------+-----------------+--------+-----------------+
+|   1000 ZEC|       3.74      |  6.01  |       8.83      |
++-----------+-----------------+--------+-----------------+
+|  10000 ZEC|      13.57      | 17.40  |      21.77      |
++-----------+-----------------+--------+-----------------+
+
+The simulation also depends on the amounts sent as specified in the next
+section. It includes the time spent waiting for the first batch to be sent.
+
+
+How much to send in each transaction
+------------------------------------
+
+The amount to send in each transaction is chosen according to the following
+distribution:
+
+1. Choose an integer exponent uniformly in the range 6 to 9 inclusive.
+2. Choose an integer mantissa uniformly in the range 1 to 99 inclusive.
+3. Calculate amount := (mantissa * 10:sup:`exponent`) zatoshi.
+4. If amount is greater than the amount remaining to send, repeat from step 1.
+
+Implementations MAY optimize this procedure by selecting the exponent and
+mantissa based on the amount remaining to avoid repetition, but the resulting
+distribution MUST be identical.
+
+Rationale
+'''''''''
+
+Suppose that a user has an amount to migrate that is a round number of ZEC.
+Then, a potential attack would be to find some subset of all the migration
+transactions that sum to a round number of ZEC, and infer that all of those
+transactions are from the same user. If amounts sent were a random multiple
+of 1 zatoshi, then the resulting knapsack problem would be likely to have a
+unique solution and be practically solvable for the number of transactions
+involved. The chosen distribution of transaction amounts mitigates this
+potential vulnerability by ensuring that there will be many solutions for sets
+of transactions, including "incorrect" solutions (that is, solutions that mix
+transactions from different users, contrary to the supposed adversary's
+inference).
+
+
+Other design decisions
```

Comment:
What about selective disclosure, and in particular payment disclosure blobs, for these transactions? Are they (experimentally...) supported as for normal transactions?

Reply:
They are basically identical to normal transactions other than how they are triggered.

Reply:
I expect so, but let's make it an explicit "should" in the ZIP.

Reply:
Added "A node SHOULD treat migration transactions in the same way as transactions
submitted over the RPC interface."